src/storage/txn/scheduler.rs

```diff
@@ -851,3 +851,109 @@ impl mio::Handler for Scheduler {
+                                     max_ts: 5,
+                                 },
+                                 Command::ResolveLock {
+                                     ctx: Context::new(),
+                                     start_ts: 10,
+                                     commit_ts: Some(20),
+                                     scan_key: None,
+                                     keys: vec![],
+                                 },
+                                 Command::Gc {
+                                     ctx: Context::new(),
+                                     safe_point: 5,
+                                     scan_key: None,
+                                     keys: vec![make_key(b"k")],
+                                 }];
+        let write_cmds = vec![Command::Prewrite {
+                                  ctx: Context::new(),
+                                  mutations: vec![Mutation::Put((make_key(b"k"), b"v".to_vec()))],
+                                  primary: b"k".to_vec(),
+                                  start_ts: 10,
+                                  options: Options::default(),
+                              },
+                              Command::Commit {
+                                  ctx: Context::new(),
+                                  keys: vec![make_key(b"k")],
+                                  lock_ts: 10,
+                                  commit_ts: 20,
+                              },
+                              Command::Cleanup {
+                                  ctx: Context::new(),
+                                  key: make_key(b"k"),
+                                  start_ts: 10,
+                              },
+                              Command::Rollback {
+                                  ctx: Context::new(),
+                                  keys: vec![make_key(b"k")],
+                                  start_ts: 10,
+                              },
+                              Command::ResolveLock {
+                                  ctx: Context::new(),
+                                  start_ts: 10,
+                                  commit_ts: Some(20),
+                                  scan_key: None,
+                                  keys: vec![make_key(b"k")],
+                              }];
+
+        let mut latches = Latches::new(1024);
+
+        let write_locks: Vec<Lock> = write_cmds.into_iter()
+            .enumerate()
+            .map(|(id, cmd)| {
+                let mut lock = gen_command_lock(&latches, &cmd);
+                assert_eq!(latches.acquire(&mut lock, id as u64), id == 0);
+                lock
+            })
+            .collect();
+
+        for (id, cmd) in readonly_cmds.iter().enumerate() {
+            let mut lock = gen_command_lock(&latches, cmd);
+            assert!(latches.acquire(&mut lock, id as u64));
+        }
+
+        // acquire/release locks one by one.
```

Comment:
IMO it would be clearer to write it in this way:
```
id: 0 => release, check unlocked
for id from 1 to max_id => acquire, release, check unlocked
```